clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -1265,6 +1329,8 @@ protected String getConfigValue(ConfigKey key, String headerName) {
                 if (key.hasDefault()) {
                     if (key.defaultValue == null)
                         return "null";
+                    if (key.overwrittenValue != null)
```

Comment:
Would this need to go before `if (key.defaultValue == null)` ? (For `state.dir` it does not matter because it does have a default, but for general purpose it seems better to change the order and make `overwrittenValue` the winning parameter if used?)

Reply:
Agree. I've changed the ordering and it will now always be the winning parameter once supplied